cps/static/js/main.js

```diff
@@ -179,13 +179,19 @@ $(document).ready(function() {
         maxVideos = maxVideos === "" ? 0 : parseInt(maxVideos);
         maxVideosSize = maxVideosSize === "" ? 0 : parseInt(maxVideosSize);
 
+        /*
         // Check if the input URL is a valid URL
         // First check if URL starts with https:// if not, prepend it
         url = url.startsWith("https://") ? url : "https://" + url;
         if (!isValidURL(url)) {
             alert("Invalid URL");
             return;
         }
+        */
+
+        // Prepend https:// if URL doesn't begin with http:// or https://
+        // As xklb requires: https://github.com/iiab/calibre-web/pull/44
+        url = /^https?:\/\//.test(url) ? url : "https://" + url;
```

Comment:
Personally, I prefer string manipulation over regex.

Reply:
To be clear:

This PR fixes support for legacy (e.g. bookmarked) URLs that begin with **http://** (i.e. without SSL / TLS).